public/components/metrics/top_menu/top_menu.tsx

```diff
@@ -139,29 +134,56 @@ export const TopMenu = ({
     </EuiButton>
   );
 
+  const savedObjectInputFromObject = (currentObject: SavedVisualization) => {
+    return {
+      ...currentObject,
+      dateRange: ['now-1d', 'now'],
+      fields: '',
+      timestamp: '@timestamp',
+    };
+  };
+
+  const updateSavedVisualization = async (metricLayout: MetricType, name: string): string => {
```

Comment:
This method written to overcome the incongruent API of OSDSavedVisualizationClient, which returns the raw saved-object on `get()` and expects a modified UI-ready object on `update()`